src/hooks/useOverflowDetection/useOverflowDetection.ts

```diff
@@ -0,0 +1,26 @@
+import { useCallback, useRef } from "react";
+
+export const useOverflowDetection = <T extends HTMLElement>() => {
+  const elementsRef = useRef<T[]>([]);
```

Comment:
suggestion: I don't think you need to keep an array of all references that can have overflow. You should use your hook within the component that can reach overflow - in your case the component for a single row

Reply:
additionally: it can even lead to memory leaks, when I use it on a paginated table where each row has expandable text, this hook will remember every single entry